app/src/components/UI/Molecules/MultiSelectFilter/SelectFilter.tsx

```diff
@@ -50,6 +59,7 @@ const SelectFilter: React.FC = () => {
       selection
       options={options}
       className={styles.filter}
+      onChange={(e, data) => dispatch(setCuisineType(data.value as string))}
```

Comment:
I figured out the root of the issue. Your type cast here is incorrect. `data.value` is an array, not a string. You should pass it as an array and fix the parameter type of the action (`PayloadAction<string>`) to take an array.